internal/gitstream/assign.go

```diff
@@ -78,49 +80,44 @@ func (a *Assign) assignIssues(ctx context.Context) error {
 			return fmt.Errorf("error while looking for SHAs in %q: %v", *issue.Body, err)
 		}
 
+		commitAuthors := make([]*github.User, 0, len(shas))
 		for _, s := range shas {
-
-			if a.DryRun {
-				logger.Info("Dry run: skipping issue update")
-				return nil
-			}
-
-			logger.Info("Assigning issue")
-
-			var (
-				assignee string
-				intErr   error
-			)
+			logger = logger.WithValues("commit", s.String())
 			if user, err := a.UserHelper.GetCommitAuthor(ctx, s.String()); err != nil {
 				if !errors.Is(err, gh.ErrUnexpectedReply) {
-					logger.Info("WARNING: failed to get a response from Github, skipping commit",
-						"issue", *issue.Number, "error", err.Error())
-					continue
-				}
-				logger.Info("WARNING: commit author for downstream issue not found on github, picking a random assignee",
-					"issue", *issue.Number, "error", err.Error())
-				assignee, intErr = a.OwnersHelper.GetRandomApprover(owners)
-				if intErr != nil {
-					return fmt.Errorf("could not get a random owner: %v", err)
+					logger.Info("WARNING: failed to get commit author from Github", "error", err.Error())
+				} else {
+					multiErr = multierror.Append(multiErr, fmt.Errorf("failed to get commit author from Github: %v", err))
 				}
 			} else {
-				if a.OwnersHelper.IsApprover(owners, *user.Login) {
-					assignee = *user.Login
+				commitAuthors = append(commitAuthors, user)
+			}
+		}
+
+		var assigned bool
+		var errFailedToAssignIssue = errors.New("could not assign issue")
```

Comment:
This should be a global variable in this file.

Reply:
Why is that? This function is the only one using it ATM.
If in the future, a caller will need to make use of it, we can export it.

Reply:
This error was removed.